packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/renderer/layout-events/useLayoutOnBoundsChange.ts

```diff
@@ -41,41 +41,47 @@ export const useLayoutOnBoundsChange = (refreshEventPayloadId: string): UseLayou
     return changes.find((change) => isMoveFinished(change) || isResizeFinished(change));
   };
 
-  const layoutOnBoundsChange = (changes: NodeChange[], nodes: Node<NodeData, DiagramNodeType>[]): void => {
-    const change = isBoundsChangeFinished(changes);
-    if (change) {
-      const diagramToLayout: RawDiagram = {
-        nodes,
-        edges: getEdges(),
-      };
+  const isSelectionChange = (changes: NodeChange[]): boolean => {
+    return changes.every((change) => change.type === 'select');
+  };
 
-      layout(diagramToLayout, diagramToLayout, (laidOutDiagram) => {
-        nodes.map((node) => {
-          const existingNode = laidOutDiagram.nodes.find((laidoutNode) => laidoutNode.id === node.id);
-          if (existingNode) {
-            return {
-              ...node,
-              position: existingNode.position,
-              width: existingNode.width,
-              height: existingNode.height,
-              style: {
-                ...node.style,
-                width: `${existingNode.width}px`,
-                height: `${existingNode.height}px`,
-              },
-            };
-          }
-          return node;
-        });
-        setNodes(nodes);
-        setEdges(laidOutDiagram.edges);
-        const finalDiagram: RawDiagram = {
-          nodes: nodes,
-          edges: laidOutDiagram.edges,
+  const layoutOnBoundsChange = (changes: NodeChange[], nodes: Node<NodeData, DiagramNodeType>[]): void => {
+    if (!isSelectionChange(changes)) {
```

Comment:
Selection changes were already filtered. You don't need to add such statement.
However, Reactflow sends a "move finished" along with the selection change, and thus, triggers the layout.
This could be fixed using `nodeDragThreshold` after upgrading to Reactflow 11.9.0.

Reply:
Updated with `nodeDragThreshold `